src/Widgets/NotificationEntry.vala

```diff
@@ -115,6 +139,13 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
         });
 
         notification.closed.connect (() => clear ());
+
+        carousel.page_changed.connect ((index) => {
+            if (index == 1) {
+                nlist.remove_entry (this);
```

Comment:
Instead of creating a circular dependency with the NotificationsList, add a "removed" signal to this class and let the list listen to this signal and remove the notification.